pylint/config/arguments_manager.py

```diff
@@ -71,6 +71,8 @@ def __init__(
             usage=usage or "%(prog)s [options]",
             description=description,
             formatter_class=_HelpFormatter,
+            # Needed to let 'pylint-config' overwrite the -h command
```

Comment:
Shoudl'nt ``pylint-config`` be enitrely independant from pylint ? In my mind it would remove code and complexity from ``pylint``. This seem to be a little too "smart" and potentially hard to understand in 6 months.

Reply:
I don't think so. `pylint-config` heavily depends on `pylint`: we need access to message definitions to check for `bad-option-value`, we need access to plugins to load them and see if they are correct, we need access to options definitions to get the default value, etc.

You can see my implementation in https://github.com/PyCQA/pylint/pull/6578. Basically, we set up as if we are in a normal run of `pylint` to make sure we have access to everything we would normally have access to, but there are 2 changes: 1) before doing the configuration parsing we add some `pylint-config` specific options and subparsers (this removes the need to do `pylint run` but allows `pylint-config generate`) and 2) we shortcircuit into the `pylint-config` handler just before we start checking files.